### ISSUE TITLE
[Bug][compiler-v2] false positive `cannot immutably borrow since mutable references exist` error in the test `imm_borrow_loc_trivial` and `imm_borrow_on_mut`

### ISSUE THREAD

**Issue by rahxephon89** [aptos-labs/aptos-core#13489]
# 🐛 Bug

When testing the test case `imm_borrow_loc_trivial` and `imm_borrow_on_mut ` in V1 test using V2, the error `cannot immutably borrow since mutable references exist` is raised while V1 does not raise any error.

**Comment by rahxephon89** [comment]
Also happened on the test case `mutable_eq_and_neq_invalid`

**Comment by rahxephon89** [comment]
The test `third_party/move/move-compiler/tests/move_check/translated_ir_tests/move/borrow_tests/imm_borrow_loc_valid.move` has the same error

**Comment by rahxephon89** [comment]
This is another code leads to the same compiler error:

```
module 0x815::m {

    enum CommonFields has drop {
    Foo{x: u64, y: u8},
    Bar{x: u64, y: u8, z: u32}
    }

    fun t9_common_field(): u64 {
        let common = CommonFields::Bar {
            x: 30,
            y: 40,
            z: 50
        };
        common.x = 15;
        common.x
    }
}
```

```
error: cannot immutably borrow since mutable references exist
   ┌─ tests/reference-safety/immu_borrow_bug.move:15:9
   │
14 │         common.x = 15;
   │         --------
   │         │
   │         used by field borrow
   │         previous mutable local borrow
15 │         common.x
   │         ^^^^^^--
   │         │
   │         requirement enforced here
   │         immutable borrow attempted here
```

**Comment by rahxephon89** [comment]
fixed by #14276
